common/src/test/java/com/microsoft/identity/common/internal/commands/GenerateShrCommandTest.java

```diff
@@ -0,0 +1,100 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.commands;
+
+import com.microsoft.identity.common.internal.authscheme.PopAuthenticationSchemeInternal;
+import com.microsoft.identity.common.internal.cache.MapBackedPreferencesManager;
+import com.microsoft.identity.common.internal.commands.parameters.CommandParameters;
+import com.microsoft.identity.common.internal.commands.parameters.GenerateShrCommandParameters;
+import com.microsoft.identity.common.internal.controllers.BaseController;
+import com.microsoft.identity.common.internal.util.ClockSkewManager;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GenerateShrCommandTest {
+
+    public static final CommandCallback EMPTY_CALLBACK = new CommandCallback() {
+        @Override
+        public void onCancel() {
+
+        }
+
+        @Override
+        public void onError(Object error) {
+
+        }
+
+        @Override
+        public void onTaskCompleted(Object o) {
+
+        }
+    };
+
+    @Test
+    public void testMappability() throws Exception {
```

Comment:
are we just testing here that we can successfully put two different command objects in a hashmap? Do we instead want to test that we can put these in a `Set` and write a test with two different command objects that are essentially equal?

Reply:
Is there a difference between a HashSet and a HashMap?

Reply:
Depends on what we are putting in for the `values` in case of a `Map`...it seems like in your case there is no difference as you are just putting a `boolean` which I believe indicates whether a key is present or not....which is why I am thinking that this should be a `HashSet` as opposed to a `Map`...because it seems like all we care about here is the size of the collection...is that the right understanding?

Reply:
Right.  It doesn't really matter to me.  The bug we're dealing with was a problem with usage of a Map, so I used a Map in the test.  And since a Set is just a Map underneath, I thought it would be clearer that what we're trying to check is that we can put things in a map and it behaves reasonably.

Reply:
I think we also want to have some test for the `equals` method of the `GenerateShrCommand` object i.e. if we create two different objects each with the same parameters such that we expect them to be equal and then we call the `equals` method on this, does it actually return true.

Reply:
I'll add tests for equals and hash code.  We should have tests for any method that has code in it, really.

Reply:
@shahzaibj Should be in latest revision